contracts/treasury/contracts/src/ZrxTreasury.sol

```diff
@@ -0,0 +1,433 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+
+  Copyright 2021 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/contracts/src/v06/IERC20TokenV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/LibERC20TokenV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/LibBytesV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/LibSafeMathV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/errors/LibRichErrorsV06.sol";
+import "@0x/contracts-zero-ex/contracts/src/features/libs/LibSignature.sol";
+import "./IZrxTreasury.sol";
+import "./DefaultPoolOperator.sol";
+import "./IStaking.sol";
+
+
+contract ZrxTreasury is
+    IZrxTreasury
+{
+    using LibERC20TokenV06 for IERC20TokenV06;
+    using LibSafeMathV06 for uint256;
+    using LibRichErrorsV06 for bytes;
+    using LibBytesV06 for bytes;
+
+    // Immutables
+    IStaking public immutable stakingProxy;
+    bytes32 public immutable defaultPoolId;
+    uint256 public immutable votingPeriod;
+    uint256 public immutable proposalThreshold;
+    uint256 public immutable quorumThreshold;
+
+    // Storage
+    Proposal[] public proposals;
+    mapping (uint256 => mapping (address => Receipt)) public voteReceipts;
+
+    /// @dev Initializes the ZRX treasury and creates the default
+    ///      staking pool.
+    /// @param stakingProxy_ The 0x staking proxy contract.
+    /// @param weth_ The WETH token contract.
+    /// @param params Immutable treasury parameters.
+    constructor(
+        IStaking stakingProxy_,
+        IERC20TokenV06 weth_,
+        TreasuryParameters memory params
+    )
+        public
+    {
+
+        require(
+            params.votingPeriod < stakingProxy_.epochDurationInSeconds(),
+            "VOTING_PERIOD_TOO_SHORT"
+        );
+        stakingProxy = stakingProxy_;
+        DefaultPoolOperator defaultPoolOperator = new DefaultPoolOperator(
```

Comment:
I think we want to expose this guy, right?

Reply:
you can get the address from staking via the pool ID, but yeah I'll just expose it here